### DISCUSSION TITLE
--open-tracing-config-file flag doesn't work on Gatsby Cloud

### DISCUSSION THREAD

**Discussion by rudouglas** [gatsbyjs/gatsby#32734] [Ideas / Feature Requests]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Currently, the only way to enable tracing is to pass the `--open-tracing-config-file` flag as part of the `gatsby build/develop` commands.

Because Gatsby Cloud just seems to run `gatsby build` and ignores these flags in the `package.json`, it's not possible to enable tracing on Gatsby Cloud

This enables tracing by allowing a user to set `OPEN_TELEMETRY_CONFIG_FILE` instead

### Reproduction Link

https://github.com/newrelic/docs-website/pull/3260

### Steps to Reproduce

1. Add `--open-tracing-config-file {path}` to package.json "build"
2. Deploy to Gatsby Cloud

### Expected Result

Tracing to work on Gatsby Cloud

### Actual Result

Tracing not working on Gatsby Cloud

### Environment

```shell
System:
    OS: macOS 11.3.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.1 - ~/.nvm/versions/node/v14.17.1/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.17.1/bin/yarn
    npm: 7.20.1 - ~/.nvm/versions/node/v14.17.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 91.0.4472.164
    Firefox: 89.0.2
    Safari: 14.1
  npmPackages:
    gatsby: ^3.10.1 => 3.10.1 
    gatsby-image: ^3.10.0 => 3.10.0 
    gatsby-plugin-gatsby-cloud: ^2.10.0 => 2.10.0 
    gatsby-plugin-json-output: ^1.2.0 => 1.2.0 
    gatsby-plugin-manifest: ^3.10.0 => 3.10.0 
    gatsby-plugin-mdx: ^2.10.0 => 2.10.0 
    gatsby-plugin-meta-redirect: ^1.1.1 => 1.1.1 
    gatsby-plugin-offline: ^4.10.0 => 4.10.0 
    gatsby-plugin-react-helmet: ^4.10.0 => 4.10.0 
    gatsby-plugin-sharp: ^3.10.2 => 3.10.2 
    gatsby-remark-autolink-headers: ^4.7.0 => 4.7.0 
    gatsby-remark-copy-linked-files: ^4.7.0 => 4.7.0 
    gatsby-remark-images: ^5.7.0 => 5.7.0 
    gatsby-source-filesystem: ^3.10.0 => 3.10.0 
    gatsby-transformer-json: ^3.10.0 => 3.10.0 
    gatsby-transformer-remark: ^4.7.0 => 4.7.0 
    gatsby-transformer-sharp: ^3.10.0 => 3.10.0 
    gatsby-transformer-yaml: ^3.10.0 => 3.10.0 
  npmGlobalPackages:
    gatsby-cli: 3.8.0
```

### Config Flags

DEV_SSR: true,
    PRESERVE_WEBPACK_CACHE: true,
    PRESERVE_FILE_DOWNLOAD_CACHE: true,

**Comment by rudouglas** [comment]
I altered this locally to check for the env first in the build and develop commands, but i'm not sure if that's the best implementation. Docs will need to updated as well if this is implemented

https://github.com/gatsbyjs/gatsby/pull/32513/files

**Comment by pragmaticpat** [comment]
@rudouglas - while this specific issue is related to enabling tracing, it sounds like you've uncovered where you're unable to run builds with various options in Gatsby Cloud as you would run locally. And what you're identifying is that you need the ability to execute builds with options as you're able to do interactively during local development.

Do I have that right?

**Reply by rudouglas** [reply]
Yes @pragmaticpat that is correct. Enabling tracing requires the `--open-tracing-config-file` flag to be passed in the run command. This works locally but not in Gatsby Cloud currently :)
